### PR TITLE
[Doc] Add descriptions of end-to-end encryption for Python client

### PR DESCRIPTION
**Pull request by Anonymitaet**
Add docs for https://github.com/apache/pulsar/pull/9588

### CODE REVIEW

**Review by zymap — APPROVED** [review]

File: site2/docs/client-libraries-python.md [hunk 1]
```diff
@@ -288,3 +288,103 @@ class Example(Record):
     a = String()
     sub = MySubRecord()
 ```
+
+## End-to-end encryption
+
+[End-to-end encryption](https://pulsar.apache.org/docs/en/next/cookbooks-encryption/#docsNav) allows applications to encrypt messages at producers and decrypt at consumers.
```
Comment (Huanli-Meng) [inline site2/docs/client-libraries-python.md:294]:
```suggestion
[End-to-end encryption](https://pulsar.apache.org/docs/en/next/cookbooks-encryption/#docsNav) allows applications to encrypt messages at producers and decrypt messages at consumers.
```

File: site2/docs/client-libraries-python.md [hunk 2]
```diff
@@ -288,3 +288,103 @@ class Example(Record):
...
+
+### Configuration
+
+If you want to use the end-to-end encryption feature in the Python client, you need to configure `publicKeyPath` and `privateKeyPath` for both producer and consumer.
```
Comment (Huanli-Meng) [inline site2/docs/client-libraries-python.md:298]:
```suggestion
To use the end-to-end encryption feature in the Python client, you need to configure `publicKeyPath` and `privateKeyPath` for both producer and consumer.
```

**Review by Huanli-Meng — COMMENTED** [review]
leave two minor comments. PTAL

**Review by tuteng — APPROVED** [review]

**Comment by Anonymitaet** [comment]
@Huanli-Meng thanks, I've incorporated, PTAL

**Review by Huanli-Meng — APPROVED** [review]

File: site2/docs/client-libraries-python.md [hunk 3]
```diff
@@ -288,3 +288,103 @@ class Example(Record):
...
+```
+publicKeyPath: "./public.pem"
+privateKeyPath: "./private.pem"
+```
+
+### Tutorial
+
+This section provides step-by-step instructions on how to use the end-to-end encryption feature in the Python client.
+
+#### Prerequisite
```
Comment (Jennifer88huang) [inline site2/docs/client-libraries-python.md:309]: Heading is used to make the structure clear, in this case, I don't think we need a heading 4 for the prerequisite. I suggest that we use a "bold font" for prerequisite or a sentence for explain it.

File: site2/docs/client-libraries-python.md [hunk 4]
```diff
@@ -288,3 +288,103 @@ class Example(Record):
...
+
+- Pulsar Python client 2.7.1 or later 
+
+#### Step
```
Comment (Jennifer88huang) [inline site2/docs/client-libraries-python.md:313]: Remove the heading. List the steps directly.

File: site2/docs/client-libraries-python.md [hunk 5]
```diff
@@ -288,3 +288,103 @@ class Example(Record):
...
+    crypto_key_reader = pulsar.CryptoKeyReader(publicKeyPath, privateKeyPath)
+    client = pulsar.Client('pulsar://localhost:6650')
+    consumer = client.subscribe(topic='encryption', subscription_name='encryption-sub', crypto_key_reader=crypto_key_reader)
+    msg = consumer.receive()
+    print("Received msg '{}' id = '{}'".format(msg.data(), msg.message_id()))
+    consumer.close()
+    client.close()
+    ```
+
+4. Run the consumer to receive encrypted messages.
```
Comment (Jennifer88huang) [inline site2/docs/client-libraries-python.md:360]: what does "run the consumer" mean?

**Review by Jennifer88huang — COMMENTED** [review]
God job~
Left some comments.

**Review by Jennifer88huang — APPROVED** [review]

**Pull request merged**